### PR TITLE
Allow for both snake_case and CamelCase for config entries written with 'consul config write'

### PR DESCRIPTION
**Pull request by rboyer**
This also has the added benefit of fixing an issue with passing time.Duration fields through config entries.

### CODE REVIEW

**Comment by rboyer** [comment]
Note this does not handle config entry bootstrap, just `consul config write`.

File: command/config/write/config_write.go [hunk 1]
```diff
@@ -132,6 +98,124 @@ func (c *cmd) Run(args []string) int {
...
+
+	var (
+		// skipWhenPatching should contain anything that legitimately contains a
+		// slice of structs when decoded.
+		skipWhenPatching  []string
+		translateKeysDict map[string]string
+	)
+
+	switch entry.GetKind() {
+	case api.ProxyDefaults:
```
Comment (mkeeler) [inline command/config/write/config_write.go]: Does this mean we don't do any snake case translation for the proxy defaults? I may need to add it then.

Comment (rboyer) [reply command/config/write/config_write.go]: There was no prior art here, so probably no. There were no fields beyond `Config` present so unless you wanted to specially handle `Config` this seemed ok.

Comment (mkeeler) [reply command/config/write/config_write.go]: And I think mapstructure was already going to allow differences in upper vs lowercase

**Pull request merged**